Sources/FoundationExtensions/UserDefaults+Extensions.swift

```diff
@@ -0,0 +1,58 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  UserDefaults+Extensions.swift
+//
+//  Created by Nacho Soto on 11/9/22.
+
+import Foundation
+
+extension UserDefaults {
+
+    #if DEBUG
+    /// The "default" `UserDefaults` to use for the SDK.
+    ///
+    /// Moving foward, this default is `UserDefaults.revenueCatSuite`,
+    /// but existing users will continue using `UserDefaults.standard` for compatibility.
+    /// This is determined by the presence of an app user ID in `UserDefaults.standard`.
+    ///
+    /// Computed property only in `DEBUG` to be able to test it under different conditions.
+    /// In release mode, it gets cached.
+    static var `default`: UserDefaults { .computeDefault() }
+    #else
+    static let `default`: UserDefaults = .computeDefault()
+    #endif
```

Comment:
This feels like an odd workaround to solve a problem that the rest of the project solves with DI - we even pass in the userDefaults when initializing purchases. 

Not saying that doing all the DI in one place is necessarily better, but it does have the advantage of looking exactly the same in all our SDKs, whereas this would be probably impossible to achieve in most languages.

Reply:
That's a good point, we don't need to cache this because we only call it once!

Reply:
Removed!